src/store/products/products.types.ts

```diff
@@ -0,0 +1,45 @@
+// Entities
+
+export type ProductId = number;
+
+export interface Product {
+  product_id: ProductId;
+  name: string;
+  purchase_cost: number;
+}
+
+// Action types
+
+export const PRODUCTS_FETCH_SUCCESS = "@@products/FETCH_SUCCESS";
+export const PRODUCTS_FETCH_BY_ID_SUCCESS = "@@products/FETCH_BY_ID_SUCCESS";
+export const PRODUCTS_ADD_SUCCESS = "@@products/ADD_SUCCESS";
+
+// Products Api
+
+export interface ProductsFetchResponseData {
+  products: Product[];
+}
```

Comment:
Here we say the fetched data will be an array of the type 'Product' defined above.